### ISSUE TITLE
Status of the project

### ISSUE THREAD

**Issue by JulioJu** [nodejs/corepack#27]
Hi,

What is the status of this project ?

We can read on the README.md

> we're working with the Node TSC to provide Corepack by default starting from Node 15, thus ensuring that all package managers can be used with little to no friction.

But it seems not be embedded into Node 15 ?

There is some plan for Node 16 or more ? Or this project is it currently dead ? (last commit some months ago, and some CI tests have failed https://github.com/nodejs/corepack/runs/1227988760 )

Thanks in advance for your answer :-)

**Comment by arcanis** [comment]
👋 tl;dr: This project is alive, but progress is currently blocked on discussions (and decisions) from the Node TSC. Specifically:

- I've opened https://github.com/nodejs/node/pull/35398 which adds Corepack to the Node binaries. It however led to many discussions which I won't retranscribe here.

- In response, a summit regarding Node distributions was hold (https://github.com/nodejs/node/discussions/35682). The [consensus](https://github.com/nodejs/node/discussions/35682#discussioncomment-147436) was that yes, Node wanted to add support for more package managers, and that they also wanted to have a tool facilitating the install of multiple Node versions at the same time (think nvm, but builtin). However:
    - It was only about a *new, additional* type of distribution, ie not about the existing binary releases themselves; so it was still not clear at the end of this meeting what were the plans regarding the existing ways to install Node.
    - It's a long-term project, and so far it seems stuck on the next step (we were supposed to hold a meeting last month, but it got delayed and I'm not sure what to expect).

- So, given that 35398 was somewhat duplicate with the plans in 35682 (although not entirely, since 35398 affects the existing distribution channels whereas 35682 is about new ones) *and that* 35682 is currently in limbo, I decided to open a more simple PR simply adding Yarn to Node: https://github.com/nodejs/node/pull/37277. Whether to land this PR or not is still being discussed.
    - I'm of course of the opinion that yes it should, but I don't really have influence on the Node project at all. At the moment their [plan](https://github.com/nodejs/node/pull/37277#issuecomment-786250871) is to hold a survey to confirm that it would be beneficial to Node's users to include Yarn.

Unfortunately in all these words there is no clear answer to your question, but only because I don't have a clear answer to give you... I personally feel like this process shouldn't be that difficult and that there's a misunderstanding somewhere, but the dynamics in play are complex, and we really want to play by the book.

In any case, you already can use Corepack by installing it from the registry, which I know is something a few people already do. Hopefully it won't be needed in the future, but I can't give you a time estimate.

**Comment by JulioJu** [comment]
@arcanis thanks a lot for your detailed answer. 

I hope Node teem will accept at least Corepack. 

Your links and discussions are so interesting.

Maybe they could be added into the README.md to check easy the progress?

Maybe one the official Website, a note about Corepack could be added. 
  * It address a problem for those who don't want embed Yarn Berry in their repository. 
  * Furthermore, thanks Corepack, no need to installl Yarn 1.x globally (that could sound strange for newbies) . 
  *  Furthermore, if a lot of people start to use it and test it, maybe it could improve chances to be merged into Node.js .

**Comment by danoc** [comment]
Here's an update from a week ago confirming that the Node team is moving towards Corepack rather than including Yarn directly in Node: https://github.com/nodejs/node/pull/37277#issuecomment-829342324

**Comment by JulioJu** [comment]
So coooooool !!!

**Comment by arcanis** [comment]
Corepack has now been released as experimental in Node 16.9

**Comment by JulioJu** [comment]
👍🏽
